src/Model_ELBDM/ELBDM_Flag_Spectral.cpp

```diff
@@ -177,18 +179,23 @@ void Prepare_for_Spectral_Criterion(const real *Var1D, real& Cond)
             Psi += Ao[order] *  Flag_Spectral_Fo[order][i];
          }
 
-         ExtensionMass += SQR(Psi.real()) + SQR(Psi.imag());
+         ExtensionMass      += SQR(Psi.real()) + SQR(Psi.imag());
+         ExtensionCellCount += 1;
       }
    } // XYZ, k,j
 
-// compute mass ratio
-// add to safeguard against vanishing physical mass
-// should not trigger refinement for vanishing physical mass and therefore be much bigger than the typical refinement thresholds
+// compute average density ratio
+// add to safeguard against vanishing density
+// should not trigger refinement for vanishing physical density and therefore be much bigger than the typical refinement thresholds
    const real Eps = 1e-4;
 // divide by a typical mass ratio at which one might want to refine in order to make
 // values in Input__Flag_Spectral look more reasonable
    const real Normalise = 1.0e18;
 
+// compute average densities from masses to be independent of patch and extension sizes
```

Comment:
@hyschive 
Since we considered a mass ratio before and a density ratio now, both values do not change after switching from mass to density. Since the density ratio should be independent of the patch size, I think they will also work for larger patch sizes.

Reply:
@KunkelAlexander Originally, I thought the volumes of the physical and extended regions were different so the density ratio and mass ratio were also different. But I just realized that `FLAG_SPECTRAL_ND = FLU_NXT = 32` so the two volumes are actually the same. Sorry for the false alarm!